diff --git a/src/activities/categorization/MenuScreen.qml b/src/activities/categorization/MenuScreen.qml
index b1eee412..b17c29c8 100644
--- a/src/activities/categorization/MenuScreen.qml
+++ b/src/activities/categorization/MenuScreen.qml
@@ -23,7 +23,6 @@
 */
 import QtQuick 2.6
 import GCompris 1.0
-import QtGraphicalEffects 1.0
 import QtQuick.Controls 1.5
 
 import "../../core"
@@ -227,22 +226,5 @@ Image {
             Behavior on y { SpringAnimation { spring: 2; damping: 0.2 } }
         }
 
-        Rectangle {
-            id: menusMask
-            visible: false
-            anchors.fill: menuGrid
-            gradient: Gradient {
-                GradientStop { position: 0.0; color: "#FFFFFFFF" }
-                GradientStop { position: 0.92; color: "#FFFFFFFF" }
-                GradientStop { position: 0.96; color: "#00FFFFFF" }
-            }
-        }
-
-        layer.effect: OpacityMask {
-            id: activitiesOpacity
-            source: menuGrid
-            maskSource: menusMask
-            anchors.fill: menuGrid
-        }
     } // grid view close
 }
diff --git a/src/activities/crane/Crane.qml b/src/activities/crane/Crane.qml
index c6fc3e78..223081a3 100644
--- a/src/activities/crane/Crane.qml
+++ b/src/activities/crane/Crane.qml
@@ -21,7 +21,6 @@
  */
 
 import QtQuick 2.6
-import QtGraphicalEffects 1.0
 
 import "../../core"
 import "crane.js" as Activity
@@ -151,10 +150,6 @@ ActivityBase {
             rows: items.rows
             z: 1
             anchors.fill: board
-            layer.enabled: true
-            layer.effect: OpacityMask {
-                maskSource: board
-            }
             
             Repeater {
                 id: gridRepeater
@@ -305,10 +300,6 @@ ActivityBase {
             z: 1
             opacity: showGrid1.opacity
             anchors.fill: modelBoard
-            layer.enabled: true
-            layer.effect: OpacityMask {
-                maskSource: modelBoard
-            }
             Repeater {
                 id: gridRepeater2
                 model: gridRepeater.model
diff --git a/src/activities/lang/MenuScreen.qml b/src/activities/lang/MenuScreen.qml
index a84755c7..525eba8a 100644
--- a/src/activities/lang/MenuScreen.qml
+++ b/src/activities/lang/MenuScreen.qml
@@ -22,7 +22,6 @@
 *   along with this program; if not, see <http://www.gnu.org/licenses/>.
 */import QtQuick 2.6
 import GCompris 1.0
-import QtGraphicalEffects 1.0
 import QtQuick.Controls 1.5
 
 import "../../core"
@@ -219,24 +218,6 @@ Image {
             Behavior on y { SpringAnimation { spring: 2; damping: 0.2 } }
         }
 
-        Rectangle{
-            id: menusMask
-            visible: false
-            anchors.fill: menuGrid
-            gradient: Gradient {
-                GradientStop { position: 0.0; color: "#FFFFFFFF" }
-                GradientStop { position: 0.92; color: "#FFFFFFFF" }
-                GradientStop { position: 0.96; color: "#00FFFFFF"}
-            }
-        }
-
-        layer.effect: OpacityMask {
-            id: activitiesOpacity
-            source: menuGrid
-            maskSource: menusMask
-            anchors.fill: menuGrid
-        }
-
     } // grid view close
 
 }
diff --git a/src/activities/menu/Menu.qml b/src/activities/menu/Menu.qml
index c4bc4398..27312f62 100644
--- a/src/activities/menu/Menu.qml
+++ b/src/activities/menu/Menu.qml
@@ -22,7 +22,6 @@ import QtQuick 2.6
 import "../../core"
 import GCompris 1.0
 import "qrc:/gcompris/src/core/core.js" as Core
-import QtGraphicalEffects 1.0
 import QtQuick.Controls 1.5
 import QtQuick.Controls.Styles 1.4
 
@@ -487,23 +486,6 @@ ActivityBase {
                 Behavior on y { SpringAnimation { spring: 2; damping: 0.2 } }
             }
        
-            Rectangle{
-                id: activitiesMask
-                visible: false
-                anchors.fill: activitiesGrid
-                gradient: Gradient {
-                  GradientStop { position: 0.0; color: "#FFFFFFFF" }
-                  GradientStop { position: 0.92; color: "#FFFFFFFF" }
-                  GradientStop { position: 0.96; color: "#00FFFFFF"}
-                }
-            }
-       
-            layer.effect: OpacityMask {
-                id: activitiesOpacity
-                source: activitiesGrid
-                maskSource: activitiesMask
-                anchors.fill: activitiesGrid
-            }
         }
 
         Rectangle {
diff --git a/src/core/main.cpp b/src/core/main.cpp
index 3631ddb0..3838b3ca 100644
--- a/src/core/main.cpp
+++ b/src/core/main.cpp
@@ -109,9 +109,9 @@ int main(int argc, char *argv[])
     // Disable it because we already support HDPI display natively
     qunsetenv("QT_DEVICE_PIXEL_RATIO");
 
-#if defined(Q_OS_WIN)
-    QApplication::setAttribute(Qt::AA_UseSoftwareOpenGL);
-#endif
+//#if defined(Q_OS_WIN)
+    QQuickWindow::setSceneGraphBackend(QSGRendererInterface::Software);
+//#endif
 
     QApplication app(argc, argv);
     app.setOrganizationName("KDE");
