diff --git a/src/core/ReadyButton.qml b/src/core/ReadyButton.qml
index f771460a..12a0b031 100644
--- a/src/core/ReadyButton.qml
+++ b/src/core/ReadyButton.qml
@@ -36,10 +36,14 @@ Rectangle {
     property variant themes: {
         "dark": {
             borderColor: "#FF373737",
+            fillColor0: "#A8FFFFFF",
+            fillColor1: "#68FFFFFF",
             textColor: "#FF373737"
         },
         "light": {
             borderColor: "white",
+            fillColor0: "#42FFFFFF",
+            fillColor1: "#23FFFFFF",
             textColor: "white"
         }
     }
@@ -57,8 +61,8 @@ Rectangle {
     width: iamReadyText.width + 50 * ApplicationInfo.ratio
     height: iamReadyText.height + 50 * ApplicationInfo.ratio
     gradient: Gradient {
-            GradientStop { position: 0 ; color: "#42FFFFFF" }
-            GradientStop { position: 1 ; color: "#23FFFFFF" }
+            GradientStop { position: 0 ; color: themes[theme].fillColor0 }
+            GradientStop { position: 1 ; color: themes[theme].fillColor1 }
         }
     signal clicked
 
